### PR TITLE
Added user status field in UserProject

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: payment_system/models.py [hunk 1]
```diff
@@ -91,10 +101,23 @@ class Project(DataOceanModel):
                                            related_name='projects')
 
     def add_user(self, user):
-        self.users.add(user)
+        UserProject.objects.create(
+            project=self,
+            user=user,
+        )
```
Comment (roman-tiukh) [inline payment_system/models.py:107]: `self.user_projects.create(user=user)`

----
`self.users.add(user)` does not work now?

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: payment_system/models.py [hunk 2]
```diff
@@ -14,10 +14,20 @@ class UserProject(DataOceanModel):
         (PARTICIPANT, 'Participant'),
     ]
 
+    INVITED = 'invited'
+    ACTIVE = 'active'
+    REMOVED = 'removed'
+    STATUSES = [
+        (INVITED, 'Invited'),
+        (ACTIVE, 'Active'),
+        (REMOVED, "Removed"),
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:23]: I think removed is better)

File: payment_system/models.py [hunk 3]
```diff
@@ -91,10 +101,23 @@ class Project(DataOceanModel):
...
+
+    def confirm_invitation(self, user):
+        user_project = UserProject.objects.get(project=self, user=user)
+        user_project.status = UserProject.ACTIVE
+        user_project.save(update_fields=['status'])
+        # should I add sending email here?
 
     def remove_user(self, user):
-        self.users.remove(user)
+        user_project = UserProject.objects.get(project=self, user=user)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:117]: done

File: payment_system/models.py [hunk 4]
```diff
@@ -91,10 +101,23 @@ class Project(DataOceanModel):
...
     def add_user(self, user):
-        self.users.add(user)
+        UserProject.objects.create(
+            project=self,
+            user=user,
+        )
+        # should I add sending email here?
+
+    def confirm_invitation(self, user):
+        user_project = UserProject.objects.get(project=self, user=user)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:111]: done

File: payment_system/models.py (hunk 1 above)
Comment (IegorSoboliev) [reply payment_system/models.py:107]: No. We can implement it like that:

user.user_projects.create(project=self)

But I think the first version looks more clear

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**